### PR TITLE
dialects: (Arith) Add custom parsing for trunci, extui, extsi

### PR DESCRIPTION
**Pull request by ajrichins**
Added more consistent parsing/printing of "simple" syntax for arith ops trunci, extui and extsi 
@math-fehr @superlopuh  @Hatsunespica

### CODE REVIEW

**Comment by ajrichins** [comment]
now passes with new lit test, xdsl-opt %s --print-op-generic --split-input-file | xdsl-opt --split-input-file | filecheck %s.  Not sure if there is a way to smoothly transition between partial simple syntax and fully simple, but willing to get that done if that's wanted

File: xdsl/dialects/arith.py [hunk 1]
```diff
@@ -775,6 +794,33 @@ class ExtSIOp(IRDLOperation):
     def __init__(self, op: SSAValue | Operation, target_type: IntegerType):
         return super().__init__(operands=[op], result_types=[target_type])
 
+    def verify_(self) -> None:
+        assert isinstance(self.input.type, IntegerType)
+        assert isinstance(self.result.type, IntegerType)
```
Comment (webmiche) [inline xdsl/dialects/arith.py:799]: This is already encoded in the field definitions, right?

File: tests/filecheck/dialects/arith/arith_ops.mlir [hunk 2]
```diff
@@ -158,4 +158,4 @@
 
   // CHECK-NEXT: %selecti = "arith.select"(%lhsi1, %lhsi32, %rhsi32) : (i1, i32, i32) -> i32
   // CHECK-NEXT: %selectf = "arith.select"(%lhsi1, %lhsf32, %rhsf32) : (i1, f32, f32) -> f32
-}) : () -> ()
+}) : () -> ()
```
Comment (webmiche) [inline tests/filecheck/dialects/arith/arith_ops.mlir:161]: add new line :)

File: xdsl/dialects/arith.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/dialects/arith.py:799]: This is for pyright reasons

**Review by superlopuh — APPROVED** [review]
thank you

**Review by tobiasgrosser — APPROVED** [review]
This looks greast.

File: tests/filecheck/dialects/arith/arith_ops.mlir (hunk 2 above)
Comment (ajrichins) [reply tests/filecheck/dialects/arith/arith_ops.mlir:161]: thanks!

**Review by math-fehr — APPROVED** [review]
Nice! Thanks a lot!

**Comment by math-fehr** [comment]
(Just add the new line and I'll merge it)

**Pull request merged**